### DISCUSSION TITLE
Compilation

### DISCUSSION THREAD

**Discussion by DanRRRR** [epochpic/epoch#628] [General]
Thanks to recent addition to Manuals called "Quick Start" my code for the first time was compiled without 1000000000 warnings. It is important that developers track what compilers and libraries fit EPOCH best. Just the "sudo apt-get install gfortran" and then the same for openmpi  simply does not work right way! Who would find out without many many tries that for that it was necessary to add MPI this way:

sudo apt-get install gfortran
sudo apt-get install openmpi-bin openmpi-common libopenmpi-dev libgtk2.0-dev

I found that just very recently when installed Linux on a clean disk.  One more question: how update correctly my other existing gfortrans on other computers? Should they be uninstalled ? How?

Would be also great to add to Quick Start same tip with Intel compiler. If gfortran with its hiccups and conflicts still was compiling the EPOCH and the code was working, Intel in my case was not compiling at all.

**Comment by DanRRRR** [comment]
If you just install gFortran taking some current latest compiler version automatically suggested by your Linux OS ("This software does not exist but can be installed --sudo install gfortran--. Would you like to do that? ") you typically will get warnings like here, all related to mismatch of error messages. Alternatively to installing exactly the version of MPI you suggested in Getting Started would it it be possible just to fix this issue such a way that no matter what version in the future will be used we do not get such warnings anymore? 
```
../SDF/FORTRAN/src/sdf_output_cartesian_ru.f90:199:37: warning: type of ‘mpi_file_write’ does not match original declaration [-Wlto-type-mismatch]
  199 |           MPI_STATUS_IGNORE, errcode)
      |                                     ^
../SDF/FORTRAN/src/sdf_output_cartesian_ru.f90:1828:37: warning: type of ‘mpi_file_write’ does not match original declaration [-Wlto-type-mismatch]
 1828 |           MPI_STATUS_IGNORE, errcode)
      |                                     ^
../SDF/FORTRAN/src/sdf_output_cartesian_r4.f90:1820:39: note: type mismatch in parameter 7
 1820 |             MPI_STATUS_IGNORE, errcode)

**Reply by Status-Mirror** [reply]
Hey @DanRRRR,

Thanks for bringing this to our attention - I had a tricky time getting EPOCH on a clean disk too, which is why I originally made the "Getting started" guide to point out the things I did to get it working.

At this moment, I don't think addressing these issues would be a good use of our development time. We're currently transitioning to the C++ code, and these FORTRAN compiler issues are not expected to be important for this task. FORTRAN development will mostly be directed at fixing bugs which prevent users from running the code or a specific feature altogether.

Cheers,
Stuart
